CoreScriptsRoot/Modules/Server/ClientChat/ChatMain.lua

```diff
@@ -548,7 +548,16 @@ local function HandleChannelJoined(channel, welcomeMessage, messageLog)
 		end
 
 		if (welcomeMessage ~= "") then
-			channelObj:AddMessageToChannel(welcomeMessage, "WelcomeMessage")
+			welcomeMessageObject = {
```

Comment:
Can we unify these keys somewhere in some util class?
welcomeMessageObject = {
   [MessageObject.KEY_ID] = -1,
   [MessageObject.KEY_FROM_SPEAKER_NAME] = nil,
   ...
}